File: 단계별로풀어보기/16.스택,큐,덱/06.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var reader = bufio.NewReader(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

func main() {
	defer writer.Flush()

	queue := []int{}
	var numCommand int
	fmt.Fscan(reader, &numCommand)

	for i := 0; i < numCommand; i++ {
		var command string 
		var val int
		fmt.Fscan(reader, &command)

		switch command {
		case "push":
			fmt.Fscan(reader, &val)
			queue = append(queue, val)
		case "pop":
			len_queue := len(queue)
			if len_queue > 0 {
				fmt.Fprintln(writer, queue[0])
				queue = queue[1:]
			} else {
				fmt.Fprintln(writer, -1)
			}
		case "size":
			fmt.Fprintln(writer, len(queue))
		case "empty":
			if len(queue) > 0 {
				fmt.Fprintln(writer, 0)
			} else {
				fmt.Fprintln(writer, 1)
			}
		case "front":
			len_queue := len(queue)
			if len_queue > 0 {
				fmt.Fprintln(writer, queue[0])
			} else {
				fmt.Fprintln(writer, -1)
			}
		case "back":
			len_queue := len(queue)
			if len_queue > 0 {
				fmt.Fprintln(writer, queue[len_queue-1])
			} else {
				fmt.Fprintln(writer, -1)
			}
		}
	}
}

File: 단계별로풀어보기/16.스택,큐,덱/06.py
import sys
from collections import deque

input = sys.stdin.readline
print = sys.stdout.write

queue = deque()
num_command = int(input())
res = list()

for _ in range(num_command):
    command, *n = input().strip().split()
    if command == "push":
        queue.append(n[0])  
    elif command == "pop":
        res.append(queue.popleft() if queue else -1)
    elif command == "size":
        res.append(len(queue))
    elif command == "empty":
        res.append(0 if queue else 1)
    elif command == "front":
        res.append(queue[0] if queue else -1)
    elif command == "back":
        res.append(queue[-1] if queue else -1)

print("\n".join(map(str, res)))
